libres/tests/analysis/test_assert_update_viable.cpp

```diff
@@ -0,0 +1,143 @@
+#include <filesystem>
+
+#include <ert/analysis/update.hpp>
+#include <ert/enkf/analysis_config.hpp>
+#include <ert/enkf/state_map.hpp>
+#include <ert/enkf/local_updatestep.hpp>
+#include <ert/enkf/local_ministep.hpp>
+#include <ert/analysis/analysis_module.hpp>
+
+#include <catch2/catch.hpp>
+
+namespace analysis {
+bool assert_update_viable(const analysis_config_type *analysis_config,
+                          const state_map_type *source_state_map,
+                          const int ensemble_size,
+                          const local_updatestep_type *updatestep);
+}
+
+TEST_CASE("asserting_update_viable", "[analysis][prototype]") {
+
+    GIVEN("Allocated memory and default value parametrs") {
+        // alloc memory for parameters
+        analysis_config_type *analysis_config = analysis_config_alloc_default();
+        state_map_type *source_state_map = state_map_alloc();
+        local_updatestep_type *updatestep =
+            local_updatestep_alloc("name_not_important");
+        int ensemble_size = 1;
+        analysis_module_type *analysis_module =
+            analysis_module_alloc("not-important");
+        local_ministep_type *ministep =
+            local_ministep_alloc("not important", analysis_module);
+        local_updatestep_add_ministep(updatestep, ministep);
+
+        THEN("Assertion fail no active realization") {
+            REQUIRE(analysis_config_get_min_realisations(analysis_config) == 0);
+            REQUIRE(state_map_get_size(source_state_map) == 0);
+            REQUIRE(local_updatestep_get_num_ministep(updatestep) == 1);
+            REQUIRE(!analysis::assert_update_viable(
+                analysis_config, source_state_map, ensemble_size, updatestep));
+        }
+
+        WHEN("State is set to STATE_HAVE_DATA") {
+            state_map_iset(source_state_map, 0, STATE_INITIALIZED);
+            state_map_iset(source_state_map, 0, STATE_HAS_DATA);
+            REQUIRE(state_map_get_size(source_state_map) == 1);
+            REQUIRE(state_map_count_matching(source_state_map,
+                                             STATE_HAS_DATA) == 1);
+            THEN("Assertion pass") {
+                REQUIRE(analysis::assert_update_viable(
+                    analysis_config, source_state_map, ensemble_size,
+                    updatestep));
+            }
+        }
+
+        WHEN("Any state but STATE_HAS_DATA") {
+            realisation_state_enum states[3] = {
+                STATE_INITIALIZED, STATE_LOAD_FAILURE, STATE_PARENT_FAILURE};
+            for (auto state : states) {
+                state_map_iset(source_state_map, 0, STATE_INITIALIZED);
+                state_map_iset(source_state_map, 0, state);
+                REQUIRE(state_map_get_size(source_state_map) == 1);
+                REQUIRE(state_map_count_matching(source_state_map, state) == 1);
+                THEN("Assertion fail no active realization") {
```

Comment:
I'm having a bit of trouble understanding this test.
It says that WHEN("Any state but STATE_HAS_DATA") but fails because there are no active realizations.
I can't see the connection between the state and the number of active realizations.

Reply:
the validation function only counts realizations that has data, thus iff `STATE_HAS_DATA` any other state should result in a 0 and zero should always equate to a `false`, One thing I don't agree with is the stderr message